src/org/opendatakit/aggregate/parser/BaseFormParserForJavaRosa.java

```diff
@@ -111,12 +116,29 @@ private static final void initializeJavaRosa() {
              // new CoreModelModule().registerModule();
              // replace with direct call to PrototypeManager
              PrototypeManager.registerPrototypes(SERIALIABLE_CLASSES);
+             redirectOutput();
              new XFormsModule().registerModule();
              isJavaRosaInitialized = true;
        }
     }
   }
 
+  private static void redirectOutput() {
+    File jrLogFile = new File(FileSystemUtils.getBriefcaseFolder(), ".briefcase-javarosa.log");
+    try {
+      final PrintStream jrOut = new PrintStream(jrLogFile);
+      Std.setOut(jrOut);
+      Std.setErr(jrOut);
+      Runtime.getRuntime().addShutdownHook(new Thread() {
```

Comment:
> Is it really the OS that closes the file, rather than the JVM?
It depends on how the JVM exits (and exits successfully).
> In any case I leave it to you whether to keep the hook.
So, I think you're right. I was being a little pedantic here. Even if someone embedded the class, this would leak file descriptors since the shutdown hook would only run upon normal VM exit. It adds to the noise and isn't all that useful. I'll remove it.